pandas/core/dtypes/cast.py

```diff
@@ -1494,21 +1494,32 @@ def cast_scalar_to_array(shape, value, dtype: Optional[DtypeObj] = None) -> np.n
 
     Parameters
     ----------
-    shape : tuple
+    shape : tuple or int
     value : scalar value
     dtype : np.dtype, optional
         dtype to coerce
 
     Returns
     -------
-    ndarray of shape, filled with value, of specified / inferred dtype
+    ndarray of shape of list of length shape[1] of Extension Arrays of length shape[0],
+        filled with value, of specified / inferred dtype
 
     """
     if dtype is None:
-        dtype, fill_value = infer_dtype_from_scalar(value)
+        dtype, fill_value = infer_dtype_from_scalar(value, pandas_dtype=True)
     else:
         fill_value = value
 
+    # TODO: Update this function to add support for 3rd party extension types
```

Comment:
why is this a TODO? isn't this exactly what this PR is doing?

Reply:
Not actually too sure on this one, @TomAugspurger pointed out that the code may not support 3rd party extension types.

But no, that is not the purpose of this PR. The purpose of this PR is to support 1st part extension types (i.e. Period, Timestamp, Interval, etc)

Reply:
The linked issue has more information. But the summary is that `infer_dtype_from_scalar` doesn't work for arbitrary scalars that might be contained in a 3rd party extension type.